### PR TITLE
Add cypress tests

### PR DESCRIPTION
**Pull request by Sakala85**

### CODE REVIEW

**Review by DonTouzi — APPROVED** [review]

**Comment by finnwoelm-sdsn** [comment]
@Sakala85 Can you rebase this on main? I think some tests will fail after rebase, because we changed the initial selection of countries in the explorerStore.

**Comment by Sakala85** [comment]
> @Sakala85 Can you rebase this on main? I think some tests will fail after rebase, because we changed the initial selection of countries in the explorerStore.
> 
> (Otherwise, OK :))

Done

File: cypress/integration/explorer/slug.js [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+      });
+
+      context("when changing background to World", () => {
+        it("renders lines for all 192 countries with data", () => {
+          cy.contains("div", "None").click();
+          cy.contains("li", "World").click();
+          it("renders lines for all 192 countries", () => {
+            cy.get('path[stroke="#d3d3d3"]').should("have.length", 192);
+          });
+        });
```
Comment (finnwoelm-sdsn) [inline cypress/integration/explorer/slug.js:40]: Why is this 192? We have 193 countries, right?

File: cypress/integration/explorer/slug.js [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+          cy.get(".legendItem").should("have.length", 11);
+          cy.get("#legend").contains("Germany");
+          cy.get("#legend").contains("United States");
+          cy.get(".MuiDrawer-docked").within(() => {
+            cy.contains("div", "Germany")
+              .prev()
+              .then((elem) => {
+                const color = elem.css("background-color");
+
+                console.log(color);
```
Comment (finnwoelm-sdsn) [inline cypress/integration/explorer/slug.js:101]: Test missing :P

File: cypress/integration/explorer/slug.js [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+        // cy.get(".MuiDrawer-docked").within(() => {
+        //   cy.contains("div", "China")
+        //     .prev()
+        //     .then((elem) => {
+        //       const color = elem.css("background-color");
+
+        //       console.log(color);
+        //     });
+        // });
```
Comment (finnwoelm-sdsn) [inline cypress/integration/explorer/slug.js:23]: Remove?

File: cypress/integration/explorer/slug.js (hunk 1 above)
Comment (finnwoelm-sdsn) [reply cypress/integration/explorer/slug.js:40]: Ah, one country doesn't have data?

File: cypress/integration/explorer/slug.js (hunk 2 above)
Comment (Sakala85) [reply cypress/integration/explorer/slug.js:101]: Yes I didn't succeed :( I keep the code here but will remove

File: cypress/integration/explorer/slug.js (hunk 1 above)
Comment (Sakala85) [reply cypress/integration/explorer/slug.js:30]: yes if no data country not shown

Comment (finnwoelm-sdsn) [reply cypress/integration/explorer/slug.js:30]: This test had `it` within another `it`, which is not allow. So it was never testing that 192 lines are rendered. In reality it's only 183 gray lines. Fixing test now. 🙃

File: cypress/integration/explorer/slug.js (hunk 2 above)
Comment (finnwoelm-sdsn) [reply cypress/integration/explorer/slug.js]: You were quite close! :)

Here is one way to do it – I just added it:

```js
        it("Have the correct Legend colors", () => {
          cy.get(".legendItem").should("have.length", 11);
          cy.get("#legend").contains("Germany");
          cy.get("#legend").contains("United States");

          let color = null;

          // get color from drawer
          cy.get(".MuiDrawer-docked").within(() => {
            cy.contains("div", "Germany")
              .prev()
              .then((elem) => {
                color = elem.css("background-color");
              });
          });

          // verify color in legend matches
          cy.get("#legend").within(() => {
            cy.contains("p", "Germany")
              .prev()
              .should("have.css", "color", color);
          });
        });
```

**Review by finnwoelm-sdsn — APPROVED** [review]

**Pull request merged**